src/driver.ts

```diff
@@ -281,6 +282,10 @@ export abstract class CMakeDriver implements vscode.Disposable {
       log.info('Removing ', cmake_files);
       await fs.rmdir(cmake_files);
     }
+    if (await fs.exists(test_file)) {
+      log.info('Removing ', test_file);
+      await fs.unlink(test_file);
```

Comment:
ctest creates a directory called Testing. Would it make sense to remove that directory instead of the xml file?

Reply:
Yes, I think that makes more sense.  Though I have another concern with this change which is that it will clear the test results for every configure, including if you simply reopened the folder in a new instance of VS Code and nothing has actually changed in the configuration.  I was originally under the impression that this was just going to happen when Kits or the variant changed.

If there's no way to tell the difference between the two (sorry, I'm new here and haven't ramped up on all the code yet), then this functionality should be regulated by a new setting so that developers can opt in to this behavior.

Reply:
Man, I really missed my mark when I did the original analysis. That side effect is quite unintended. I propose that I drop all the changes to src/driver.ts at this point, keeping the change to src/cmake-tools.ts which makes the "Run CTest" button appear when there are tests available in the current configuration.

The test result handling can be made into a new issue, where other options regarding the handling of test results can be discussed as well.